src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs

```diff
@@ -193,6 +198,35 @@ protected override StatementSyntax AsEmbeddedStatement(IEnumerable<StatementSynt
                 : updatedIf.WithAdditionalAnnotations(Formatter.Annotation);
         }
 
+        private static bool ShouleKeepFalse(IfStatementSyntax ifStatement, StatementSyntax falseStatement)
+        {
+            if (falseStatement is BlockSyntax falseBlock)
+            {
+                // Block false syntax with some statements included.
+                // If there are no statements, it's an empty
+                // block and we can get rid of it.
+                if (falseBlock.Statements.Any())
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                // The statement is not expected to have children, so we know it's fine
+                // to consider this something that needs to be included. Such as 
+                // a return statement, or other similar things for single line if/else.
+                return true;
+            }
+
+            // If the stateements for the else don't pass, we still need to check
+            // if there are comments from the original if that should be included.
+            // If so, pass them along to be copied to the new else block
+            return ifStatement.Statement is BlockSyntax block
+                && block.CloseBraceToken.LeadingTrivia.Any(t => 
+                    t.IsKind(SyntaxKind.MultiLineCommentTrivia) ||
+                    t.IsKind(SyntaxKind.SingleLineCommentTrivia));
```

Comment:
maybe check the openbracetopen.trailing trivia as well.  Also, i think we have some IsAnyComment extensions to unify the two kind checks.

Reply:
I saw one for document comments, I'll check if I can find one for normal comments. Good idea on checking openbrace as well